gpbft/gpbft.go

```diff
@@ -918,6 +983,7 @@ func (i *instance) broadcast(round uint64, step Phase, value ECChain, createTick
 	}
 
 	_ = i.participant.host.RequestBroadcast(mb)
+	i.getRound(i.round).broadcasted.record(mb)
```

Comment:
This is fragile in the case of DECIDE messages. A DECIDE message doesn't fundamentally have a round, but is being recorded here against a particular round state. If `i.round` ever changes after this, the DECIDE message may be omitted from subsequent rebroadcasting, though it's the only message that should actually be sent after that point.

We discussed that perhaps it would be better to move the `map[round][]*MB` into the broadcastState rather than have it be spread across the round states, and then the decide message can be stored distinctly alongside that round map. And then you pointed out that we can do pruning of messages that will never need to be rebroadcast upon addition to that state, rather than any filtering while reading it.